AWS: 0 to COS (#302)
canonical/observability: terraform/modules/aws/README.md

---

## Available Commands (via `just`)

- `just init` – Initialize Terraform in both modules
- `just apply` – Provision AWS infrastructure, then pipe the necessary outputs to provision COS on top
- `just destroy` – Tear down everything (COS + infra)

## Prerequisites

Make sure you have the following installed:

- [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
- [AWS CLI](https://github.com/aws/aws-cli)
- [Juju](https://snapcraft.io/juju)
- [Just](https://github.com/casey/just)

### AWS Credentials Setup

Before running any commands, ensure your AWS credentials are configured on the host:

You can do this using one of the following methods:

- [Environment variables](https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-envvars.html)
- [Credentials file](https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-files.html)

---

## Usage

### Option 1: Full Bootstrap (Provision Infra + Deploy COS)

To go from zero to COS:

```
just apply <aws_region>
```

---

### Option 2: Deploy COS only

> **Note:** This option assumes that the required AWS infrastructure already exists and that a Juju controller has been bootstrapped.  

> > Additionally, a Juju model must also be created. 

Reviewer: ```suggestion
> Additionally, a Juju model must be ready in advance.
```

> ```

Reviewer: Add `bash` to backticks?
